src/main/java/io/cdap/e2e/utils/ElementHelper.java

```diff
@@ -0,0 +1,169 @@
+  private static final JavascriptExecutor js = (JavascriptExecutor) SeleniumDriver.getDriver();
+  private static final Actions actions = new Actions(SeleniumDriver.getDriver());
+
+  public static void scrollToElement(WebElement element) {
+    logger.info("Scrolling to the element: " + element);
+    actions.moveToElement(element).perform();
+  }
+
+  /**
+   * Scrolling to get the element in view using the Javascript Executor.
+   * It should only be used if necessary, as it does not replicate how a user would interact with the page.
+   * @param element
+   */
+  public static void scrollToElementUsingJsExecutor(WebElement element) {
+    logger.info("Scrolling to the element: " + element + " using the JsExecutor");
+    js.executeScript(ConstantsUtil.JS_SCROLL_INTO_VIEW, element);
+  }
+
+  /**
+   * Clicking on an element using the Javascript Executor.
+   * It should only be used if necessary, as it does not replicate how a user would interact with the page.
+   * @param element
+   */
+  public static void clickOnElementUsingJsExecutor(WebElement element) {
+    logger.info("Click on the element using Javascript Executor: " + element);
+    js.executeScript(ConstantsUtil.JS_CLICK, element);
+  }
+
+  public static void clickOnElement(WebElement element) {
+    WaitHelper.waitForElementToBeClickable(element);
+    scrollToElement(element);
+    logger.info("Click on the element: " + element);
+    element.click();
+  }
+
+  public static void clickOnElement(WebElement element, long timeoutInSeconds) {
+    WaitHelper.waitForElementToBeClickable(element, timeoutInSeconds);
+    scrollToElement(element);
+    logger.info("Click on the element: " + element);
+    element.click();
+  }
+
+  public static void clickIfDisplayed(WebElement element, long timeoutInSeconds) {
+    if (isElementDisplayed(element, timeoutInSeconds)) {
+      clickOnElement(element, timeoutInSeconds);
+    }
+  }
+
+  public static void clickIfDisplayed(WebElement element) {
+    clickIfDisplayed(element, ConstantsUtil.SMALL_TIMEOUT_SECONDS);
+  }
+
+  public static void sendKeys(WebElement element, String keys) {
+    WaitHelper.waitForElementToBeDisplayed(element);
+    scrollToElement(element);
+    logger.info("Send keys to element: " + element);
+    element.sendKeys(keys);
+  }
+
+  public static void hoverOverElement(WebElement element) {
+    WaitHelper.waitForElementToBeDisplayed(element);
+    scrollToElement(element);
+    logger.info("Hovering over the element: " + element);
+    actions.moveToElement(element).build().perform();
+  }
+
+  public static void clearElementValue(WebElement element, int limit) {
+    WaitHelper.waitForElementToBeDisplayed(element);
+    scrollToElement(element);
+    logger.info("Clearing WebElement: " + element);
+    boolean flag = false;
+    element.click();
+    element.clear();
+
+    for (int i = 0; i < limit; i++) {
+      if (element.getAttribute("value").equals("")) {
+        flag = true;
```

Comment:
nit: no need for a flag here, just return if element.getAttribute("value").equals(""). Assert.fail if we come out of the for loop

Reply:
Updated